src/dataviews/dataviews-factory.js

```diff
@@ -69,12 +81,17 @@ module.exports = Model.extend({
   },
 
   _indexOf: function (layerModel) {
-    var index = this._interactiveLayersCollection.indexOf(layerModel);
+    // We need to filter the layers to only select those that Windshaft knows about
+    // and be able to calculate the right index.
+    var interactiveLayers = this._layersCollection.select(function (layer) {
+      return layer.get('type') === 'CartoDB' || layer.get('type') === 'torque';
+    });
+
+    var index = interactiveLayers.indexOf(layerModel);
```

Comment:
I was reading up on _why_ we would need the layer here, I think we could :fire: both this change and this entire filter-layer reference. 

But, instead of writing several paragraphs here, let's do a quick hangout to discuss it, let's summarize the decision afterwards if it makes sense.